spec/requests/api/notifications_spec.rb

```diff
@@ -24,11 +25,53 @@
   end
 
   describe 'notification delete' do
-    it 'is not supported' do
-      api_basic_authorize
+    context 'on resource' do
+      it 'deletes notification using POST' do
+        api_basic_authorize
+
+        run_post(notification_url, gen_request(:delete))
+        expect(response).to have_http_status(:ok)
+        expect_single_action_result(:success => true, :href => :notification_url)
+        expect { notification_recipient.reload }.to raise_error(ActiveRecord::RecordNotFound)
+      end
+
+      it 'deletes notification using DELETE' do
+        api_basic_authorize
+
+        run_delete(notification_url)
+        expect(response).to have_http_status(:no_content)
+        expect { notification_recipient.reload }.to raise_error(ActiveRecord::RecordNotFound)
+      end
+    end
+
+    context 'on collection' do
+      it 'rejects on notification that is not owned by current user' do
+        api_basic_authorize
+
+        run_post(notifications_url, gen_request(:delete, :href => foreign_notification_url))
+        expect(response).to have_http_status(:not_found)
```

Comment:
Yes, any collection that implements `find_XXX` method behaves like this.

See https://github.com/ManageIQ/manageiq/blob/c80772d6d801d1cd6951c372d9f044ccea8f767a/app/controllers/api/base_controller/renderer.rb#L182

We could fix this, but we would need to rewrite all the `find_XXX` methods with `custom_filtered_XXX` methods.